vscode/src/svelte/components/file/File.svelte

```diff
@@ -0,0 +1,102 @@
+<script lang='ts'>
+  import Fa from "svelte-fa";
+  import { faCaretRight } from "@fortawesome/free-solid-svg-icons";
+
+  import { FileInclusionStatus } from "../../../types/globals";
+  import { FileTreeElement } from "./FileTreeElement";
+  import File from './File.svelte';
+  import Checkbox from "./Checkbox.svelte";
+  
+  export let file: FileTreeElement;
+  export let indent: number = 0;
+  let status: FileInclusionStatus;
+  file.statusStore.subscribe(value => status = value);
+
+  let isOpen: boolean = false;
+  const handleClickRow = () => {
+    if (file.file.children) {
+      isOpen = !isOpen;
+    } else {
+      file.handleClick();
+    }
+  }
+  const handleClickCheckbox = () => {
+    file.handleClick();
+  }
+  
+  let _faded: boolean = false;
+  $: _faded = status === FileInclusionStatus.autoIncluded || status === FileInclusionStatus.autoExcluded;
+
+  // Special case where checkbox renders differently
+  let renderMixedChildren: boolean = false;
+  $: renderMixedChildren = !isOpen && file.hasMixedChildren && !!status;
+    
+</script>
+
+<div class="file-group">
+  <!-- svelte-ignore a11y-click-events-have-key-events -->
+  <div 
+    class="file-line {_faded ? 'faded' : ''}" 
+    on:click={handleClickRow}
+    role="button"
+    tabindex="0"
+  >
+    <div style={`width: ${indent}em`} />
+    {#if file.children}
+      <span class="caret {isOpen ? 'open' : ''}" >
+        <Fa icon={faCaretRight} />
+      </span>
+    {/if}
+    <Checkbox 
+      status={status} 
+      renderMixedChildren={renderMixedChildren}
+      handleClick={handleClickCheckbox} 
+    />
+    <span class="file-name">{file.file.name}</span>
+  </div>
+  
+  <div class="children-container {isOpen ? 'open' : ''}">
+    {#if file.children}
```

Comment:
I notice a ~1 second latency when toggling expand/contract for large directories which can be fixed by checking `isOpen` here.